### PR TITLE
284 document improvement

### LINKED ISSUE #284: Create consistent documentation using the NatSpec format for contracts, functions, modifiers, events and state variable declarations
## Issue Type 

```
[x ] feature request
```

## Current Behavior 
A lot of parts of the contract system are not documented. 

## Expected Behavior
In some parts we started to use the NatSpec format for example here: https://github.com/omisego/plasma-contracts/blob/master/plasma_framework/contracts/src/framework/ExitGameController.sol#L65-L74

I propose that we create documentation for all contracts, functions, modifiers, events and state variable declarations across the entire contract system.

### PR DESCRIPTION
**Pull request by boolafish**
closes #284 (last part of this)

### CODE REVIEW

**Comment by thec00n** [comment]
Sorry to just push my changes into this PR @boolafish. I was a lot faster working through the contracts in the folders. Could you have a look and make sure you are ok with the changes? 

Also please find some more questions and feedback below:

Could we consider changing the parameter _target to _transferee and owner to transferor?
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/vaults/Erc20Vault.sol#L45-L55https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/vaults/EthVault.sol#L27-L42

We should add documentation here:
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/vaults/verifiers/Erc20DepositVerifier.sol#L17https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/vaults/verifiers/EthDepositVerifier.sol#L15

Could we rename this function into sth for descriptive
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/utils/IsDeposit.sol#L11-L13

What does that mean and what is a wire format?
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/transactions/WireTransaction.sol#L23-L25

Link here to something that explains the periods?
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/framework/PlasmaFramework.sol#L13-L14

Document more:
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/framework/ExitGameController.sol#L63

File: plasma_framework/contracts/src/vaults/Erc20Vault.sol [hunk 1]
```diff
@@ -30,27 +30,27 @@ contract Erc20Vault is Vault {
...
 
         emit DepositCreated(msg.sender, blknum, token, amount);
     }
 
     /**
     * @notice Withdraw ERC20 tokens that have been exited from the OMG network successfully.
-    * @param _target address of the transferee
-    * @param _token address of ERC20 token contract.
-    * @param _amount amount to transfer.
+    * @param transferee address of the transferee
```
Comment (thec00n) [inline plasma_framework/contracts/src/vaults/Erc20Vault.sol]: Should be transferor?

**Comment by boolafish** [comment]
@thec00n hi, just pushed some changes for your request.

> Could we rename this function into sth for descriptive
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/utils/IsDeposit.sol#L11-L13

well....I used the convention of Java Predicate which has the `test` function. Anyway do you have any recommendation? In places where caller uses this function it would be like this: `isDeposit.test(utxoPos[i].blockNum());` which IMO looks pretty straight forward.

> What does that mean and what is a wire format?
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/transactions/WireTransaction.sol#L23-L25

Call out again. @paulperegud how would you describe wire format......

> Add documentation to these functions. They internal/private but they contains quite dense logic and should therefore documented. 
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/framework/utils/PriorityQueue.sol
https://github.com/omisego/plasma-contracts/blob/52eace3e82a2dcd36c4aef1d2e9e7d9ae642a351/plasma_framework/contracts/src/transactions/eip712Libs/PaymentEip712Lib.sol

For EIP712 one: I added some more links more specifically mapping with the function. Though the page was linked already on top of the contract.

For Priority queue one, I can hardly agree to add comment on those private functions. Priority queue is really a common sense and the implementation there is really standard heap IMO. If you think it does not express itself enough, let's try to give better function names/ variables instead.

File: plasma_framework/contracts/src/vaults/Vault.sol [hunk 2]
```diff
@@ -69,4 +68,18 @@ contract Vault is Operated {
             return depositVerifiers[0];
         }
     }
+
+    /**
+     * @dev Function that should be called by the child contract inheritancing this on deposit.
```
Comment (thec00n) [inline plasma_framework/contracts/src/vaults/Vault.sol]: What does that mean?

Comment (boolafish) [reply plasma_framework/contracts/src/vaults/Vault.sol]: Is this better? 
https://github.com/omisego/plasma-contracts/pull/325/commits/ea33eea626f932f261d6b77c4b3faa529d7d3584

**Comment by thec00n** [comment]
@boolafish I think we can merge this once you are ready. If there are other parts that need improvements or more documentation then we can address this as part of another iteration.

File: plasma_framework/contracts/src/vaults/Vault.sol (hunk 2 above)
Comment (thec00n) [reply plasma_framework/contracts/src/vaults/Vault.sol]: `@dev designed to be called by the contract that inherits Vault`

**Pull request merged**